pkg/agent/openflow/pipeline.go

```diff
@@ -2060,26 +2060,30 @@ func (f *featureNetworkPolicy) dnsPacketInFlow(conjunctionID uint32) binding.Flo
 		Done()
 }
 
-// localProbeFlow generates the flow to forward locally generated packets to ConntrackCommitTable, bypassing ingress
+// localProbeFlows generates the flows to forward locally generated packets to stageConntrack directly, bypassing ingress
 // rules of Network Policies. The packets are sent by kubelet to probe the liveness/readiness of local Pods.
-// On Linux and when OVS kernel datapath is used, it identifies locally generated packets by matching the
-// HostLocalSourceMark, otherwise it matches the source IP. The difference is because:
-// 1. On Windows, kube-proxy userspace mode is used, and currently there is no way to distinguish kubelet generated
-//    traffic from kube-proxy proxied traffic.
-// 2. pkt_mark field is not properly supported for OVS userspace (netdev) datapath.
-// Note that there is a defect in the latter way that NodePort Service access by external clients will be masqueraded as
-// a local gateway IP to bypass Network Policies. See https://github.com/antrea-io/antrea/issues/280.
-// TODO: Fix it after replacing kube-proxy with AntreaProxy.
-func (f *featurePodConnectivity) localProbeFlow(ovsDatapathType ovsconfig.OVSDatapathType) []binding.Flow {
+// - On Linux and when OVS kernel datapath is used, it identifies locally generated packets by matching the HostLocalSourceMark.
+// - On Windows or OVS userspace (netdev) datapath is used, one matching condition is that the source IP is the Antrea
```

Comment:
Suggestion:

// On Windows or when OVS userspace (netdev) datapath is used, we need a different approach because:
// 1. On Windows, kube-proxy userspace mode is used, and currently there is no way to distinguish kubelet generated
//    traffic from kube-proxy proxied traffic.
// 2. pkt_mark field is not properly supported for OVS userspace (netdev) datapath.
//   - when proxyAll is disabled, the probe packets are identified by matching the source IP is the Antrea gateway IP.
//   - When proxyAll is enabled, the packets are identified by matching both the Antrea gateway IP and NotServiceCTMark.

Reply:
Thanks, updated.